README.md

```diff
@@ -0,0 +1,164 @@
+# gulp-core-build [![npm version](https://badge.fury.io/js/%40microsoft%2Fgulp-core-build.svg)](https://badge.fury.io/js/%40microsoft%2Fgulp-core-build)
```

Comment:
For the root README lets just have a summary of concept and links to the various projects. I've checked this in as a basic first step, but it could use more of the definitions. Could you merge master?